### ISSUE TITLE
Fetch completed tasks for >1 day

### ISSUE THREAD

**Issue by plewien** [Ledaryy/obsidian-todoist-completed-tasks#3]
Hey,

I like the idea of this plugin, but my current workflow is to have a "weekly" note which summarizes my completed Todoist tasks. Fetching only the past day does not suit my use-case. 

Can we add a feature that will fetch completed tasks from the past `x` amount of days?

**Comment by Ledaryy** [comment]
Hi,
Thanks for using the plugin!
I think it is possible. Let me think about the implementation and I'll get back to you.

**Comment by Ledaryy** [comment]
Hey @plewien,
I just finished developing the main logic for it.
But there are some issues with it. Sync API limits us only to 30 tasks per request. I'll try to find a turnaround, but probably it will be really difficult. And I'm not sure if is it really worth investing time in it. I'm currently reaching for Todoist support, about the limitations. I'll update you all on it.

@ddeighton yes, I'm currently using this API. It is also in use for daily logic as well.

**Comment by ddeighton** [comment]
Can you set the `limit` parameter to 200? That appears to be the max per request. Beyond that, it would take multiple requests with the `offset` parameter.

**Comment by plewien** [comment]
Having a limit of 200 tasks would be reasonable for my use-case, which would avoid concerns about pagination. I'm expecting to max out around 40 tasks for a given week with how I'm currently using Todoist.

+1 for date ranges, if possible implemented through the `%%` comments in the note. That way I can use Templater to populate the date ranges in weekly notes.

**Comment by Ledaryy** [comment]
Hey @ddeighton, @plewien 
I finished implementation, and that was a great idea to set the limit to 200.
Also, I found the limitations. In the current logic, to render the subtasks, we need to fetch every single task detail (because that is the only way to find the parent task ID). And it seems like 30 requests are the max. 
So, I've added the switch to disable the subtasks rendering, so it should work fine. The current limits will be 30 tasks with subtask rendering and 200 tasks without subtask rendering.
Also, I've added the "Fetch completed todoist tasks for last n hours" command with a prompt modal.
